### PR TITLE
Use protobuf Arenas in LockFreeBufferCaptureEventProducer

### PR DESCRIPTION
**Pull request by dpallotti**
#### Use protobuf Arenas in LockFreeBufferCaptureEventProducer
After various experimenting, I can say that this can save *a lot* of extra work
in `LockFreeBufferCaptureEventProducer::ForwarderThread`, specifically in terms
of heap allocations and deallocations.

The one downside is that it prevents moves from an already constructed protobuf,
which is the case for the Vulkan layer. I tested the two cases on Trata and
Infiltrator and didn't notice any difference: `ForwarderThread` was and still is
around 1 to 2 % CPU utilizaton.

Bug: http://b/179969217

### CODE REVIEW

File: src/OrbitProducer/include/OrbitProducer/LockFreeBufferCaptureEventProducer.h [hunk 1]
```diff
@@ -74,15 +81,34 @@ class LockFreeBufferCaptureEventProducer : public CaptureEventProducer {
...
+  //   allocated no matter what;
+  // - If `IntermediateEventT` is itself a `ProducerCaptureEvent`, or the type of one of its fields,
+  //   attempting to move from it into the Arena-allocated `ProducerCaptureEvent` will silently
+  //   result in a deep copy.
+  [[nodiscard]] virtual orbit_grpc_protos::ProducerCaptureEvent* TranslateIntermediateEvent(
+      IntermediateEventT&& intermediate_event, google::protobuf::Arena* arena) = 0;
 
  private:
   void ForwarderThread() {
+    pthread_setname_np(pthread_self(), "ForwarderThread");
```
Comment (pierricgimmig) [inline src/OrbitProducer/include/OrbitProducer/LockFreeBufferCaptureEventProducer.h:99]: orbit_base::SetCurrentThreadName("ForwarderThread"); ?

File: src/OrbitProducer/include/OrbitProducer/LockFreeBufferCaptureEventProducer.h [hunk 2]
```diff
@@ -74,15 +81,34 @@ class LockFreeBufferCaptureEventProducer : public CaptureEventProducer {
...
+
     constexpr uint64_t kMaxEventsPerRequest = 10'000;
     std::vector<IntermediateEventT> dequeued_events(kMaxEventsPerRequest);
+
+    // Pre-allocate and always reuse the first memory block for each Arena instance below in the
+    // order of magnitude of 1 MB. This is a small but measurable performance improvement.
+    google::protobuf::ArenaOptions arena_options;
+    constexpr size_t kArenaInitialBlockSize = kMaxEventsPerRequest * 128;
```
Comment (pierricgimmig) [inline src/OrbitProducer/include/OrbitProducer/LockFreeBufferCaptureEventProducer.h:107]: what is 128 here? Could we make it a constant?

File: src/Service/ProducerSideServiceImpl.cpp [hunk 3]
```diff
@@ -354,6 +354,8 @@ void ProducerSideServiceImpl::ReceiveEventsThread(
     grpc::ServerReaderWriter<orbit_grpc_protos::ReceiveCommandsAndSendEventsResponse,
                              orbit_grpc_protos::ReceiveCommandsAndSendEventsRequest>* stream,
     uint64_t producer_id, bool* all_events_sent_received) {
+  pthread_setname_np(pthread_self(), "PSSI::RcvEvents");
```
Comment (pierricgimmig) [inline src/Service/ProducerSideServiceImpl.cpp:357]: orbit_base::SetCurrentThreadName() ?

**Review by pierricgimmig — APPROVED** [review]
Hi Davide, thanks for this! I'm looking forward to the performance improvements. LGTM, but please see minor comments.

**Review by dimitry- — APPROVED** [review]

File: src/Service/ProducerSideServiceImpl.cpp (hunk 3 above)
Comment (dpallotti) [reply src/Service/ProducerSideServiceImpl.cpp]: Done.

File: src/OrbitProducer/include/OrbitProducer/LockFreeBufferCaptureEventProducer.h (hunk 1 above)
Comment (dpallotti) [reply src/OrbitProducer/include/OrbitProducer/LockFreeBufferCaptureEventProducer.h]: Done.

File: src/OrbitProducer/include/OrbitProducer/LockFreeBufferCaptureEventProducer.h (hunk 2 above)
Comment (dpallotti) [reply src/OrbitProducer/include/OrbitProducer/LockFreeBufferCaptureEventProducer.h]: I changed this to be 1 MB without involving `kMaxEventsPerRequest`, it should be less confusing.

**Review by florian-kuebler — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**